### ISSUE TITLE
🙋 Keep folder structure and filename

### ISSUE THREAD

**Issue by zeakd** [parcel-bundler/parcel#433]
Thanks for making this amazing library.

### 🤔 Expected Behavior

``` html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <title>Document</title>
</head>
<body>
  <a href="./a.html"></a>
  <script src='./index.js'></script>
  <script src='./main.js'></script>
  <script src='./b/index.js'></script>
</body>
</html>
```

Currently, all folder are flatten and filename become hashed with '/dist'. but how about keep the folder structure and filename? like,

``` html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <title>Document</title>
</head>
<body>
  <a href="/a.html"></a>
  <script src='/index.js'></script>
  <script src='/main.js'></script>
  <script src='/b/index.js'></script>
</body>
</html>
```
### 😯 Current Behavior

``` html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <title>Document</title>
</head>
<body>
  <a href="/dist/328a03237686de9183d07e1b3ecb7768.html"></a>
  <script src="/dist/example-app.js"></script>
  <script src="/dist/81b87a177e4cfca53b661600c0bf5cd2.js"></script>
  <script src="/dist/b2d29b73372f261f4f9869fe092e7d54.js"></script>
</body>
</html>
```

### 🔦 Context

#303 #323 #307 #280 are related. 

I think It have several advantages.

- easy to understand. it doesn't need many options. 
- dist folder can be served as it is 
- don't need to concern about multiple files with the same name

Maybe I missed your intentions, but in my opinion this feature make parcel more flexible.

**Comment by zeakd** [comment]
I made a PR for this. #557

**Comment by devongovett** [comment]
Should be solved by #1025 which retains filenames and folder structure for linked files, and content hashes static assets. Please help test using the master branch - a release will hopefully come next week!
